tools/comdb2ar/serialise.cpp

```diff
@@ -263,11 +267,30 @@ static void serialise_file(const FileInfo& file, volatile iomap *iomap=NULL,
 
             int retry = 5;
             ssize_t n = 0;
+
+            std::string incrFilename = incr_path + "/" + filename + ".incr";
+            std::ofstream incrFile(incrFilename,
+                                    std::ofstream::binary |
+                                    std::ofstream::trunc);
+
             while (n < bytesread && retry) {
-                if (verify_checksum(pagebuf + n, pagesize, file.get_crypto(), file.get_swapped())) {
+                bool verify_bool = false;
+                PAGE * pagep = (PAGE *) (pagebuf + n);
+                uint32_t verify_cksum;
+                verify_checksum(pagebuf + n, pagesize, file.get_crypto(), file.get_swapped(), &verify_bool, &verify_cksum);
+
+                if(verify_bool){
                     // checksum verified
                     n += pagesize;
                     retry = 5;
+
+                    // If we are in incremental mode, on initial backup creation we want to create the diff files
+                    if(incr_create){
+                        incrFile.write((char *) &(LSN(pagep).file), 4);
```

Comment:
Could you endianize these files? Test by doing a restore on ibm using linux generated increment files (and vice-versa).